### PR TITLE
ROX-17300: ACS Host aliases

### PR DESCRIPTION
**Pull request by ludydoo**
Version(s):
RHACS 4.5

Issue:
[ROX-17300](https://issues.redhat.com//browse/ROX-17300)

Link to docs preview:

[76133--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-central-config-options-ocp.html](https://76133--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-central-config-options-ocp.html)
[76133--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-config-options-ocp.html](https://76133--ocpdocs-pr.netlify.app/openshift-acs/latest/installing/installing_ocp/install-secured-cluster-config-options-ocp.html)

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by GrimmiMeloni — APPROVED** [review]
Looks good. I **_assume_** the change from `scanner.Tolerations` to `scanner.analyzer.tolerations` is expected.

**Comment by ludydoo** [comment]
> Looks good. I **_assume_** the change from `scanner.Tolerations` to `scanner.analyzer.tolerations` is expected.

@GrimmiMeloni yes, I noticed the mistake as I was adding the new fields, figured I'd include the fix here

File: modules/central-configuration-options-operator.adoc [hunk 1]
```diff
@@ -31,6 +31,9 @@ Otherwise, you must reinstall the custom resource to log back in.
 | `central.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Central. This parameter is mainly used for infrastructure nodes.
 
+| `central.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:35]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 2]
```diff
@@ -95,6 +98,9 @@ This parameter has the following constraints:
 | `central.db.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Central DB. This parameter is mainly used for infrastructure nodes.
 
+| `central.db.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:102]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 3]
```diff
@@ -129,6 +135,9 @@ If no PVC with the given name exists, it is created. The default value is `centr
 | `scanner.analyzer.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for the StackRox Scanner. This parameter is mainly used for infrastructure nodes.
 
+| `scanner.analyzer.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:139]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 4]
```diff
@@ -183,6 +195,9 @@ include::snippets/technology-preview.adoc[]
 | `scannerV4.db.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Scanner V4 DB. This parameter is mainly used for infrastructure nodes.
 
+| `scannerV4.db.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:199]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 5]
```diff
@@ -199,6 +214,9 @@ If no PVC with the given name exists, it is created. The default value is `scann
 | `scannerV4.indexer.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for the Scanner V4 Indexer. This parameter is mainly used for infrastructure nodes.
 
+| `scannerV4.indexer.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:218]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 6]
```diff
@@ -223,6 +241,9 @@ If no PVC with the given name exists, it is created. The default value is `scann
 | `scannerV4.matcher.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for the Scanner V4 Matcher. This parameter is mainly used for infrastructure nodes.
 
+| `scannerV4.matcher.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:245]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 7]
```diff
@@ -53,6 +53,9 @@ The default value is `false`.
 | `admissionControl.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Admission Control. This parameter is mainly used for infrastructure nodes.
 
+| `admissionControl.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/secured-cluster-configuration-options-operator.adoc:57]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 8]
```diff
@@ -92,6 +95,12 @@ Use Scanner configuration settings to modify the local cluster scanner for the O
 | `scanner.analyzer.nodeSelector`
 | Specify a node selector label as `label-key: label-value` to force Scanner to only schedule on nodes with the specified label.
 
+| `scanner.analyzer.tolerations`
+| If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Scanner.
+
+| `scanner.analyzer.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/secured-cluster-configuration-options-operator.adoc:102]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 9]
```diff
@@ -116,12 +125,12 @@ Use Scanner configuration settings to modify the local cluster scanner for the O
...
 | The default number of replicas. The default value is `3`.
 
-| `scanner.Tolerations`
-| If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Scanner.
-
 | `scanner.db.nodeSelector`
 | Specify a node selector label as `label-key: label-value` to force Scanner DB to only schedule on nodes with the specified label.
 
+| `scanner.db.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/secured-cluster-configuration-options-operator.adoc:132]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 10]
```diff
@@ -194,6 +203,9 @@ If you use the `Slim` image type, you must ensure that your Central instance is
 | `perNode.compliance.resources.limits`
 | Use this parameter to override the default resource limits for Compliance.
 
+| `perNode.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/secured-cluster-configuration-options-operator.adoc:207]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/secured-cluster-configuration-options-operator.adoc [hunk 11]
```diff
@@ -224,6 +236,9 @@ This configuration defines the settings of the Sensor components, which runs on
 | `sensor.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for Sensor. This parameter is mainly used for infrastructure nodes.
 
+| `sensor.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/secured-cluster-configuration-options-operator.adoc:240]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

File: modules/central-configuration-options-operator.adoc [hunk 12]
```diff
@@ -153,6 +162,9 @@ If no PVC with the given name exists, it is created. The default value is `centr
 | `scanner.db.tolerations`
 | If the node selector selects tainted nodes, use this parameter to specify a taint toleration key, value, and effect for the StackRox Scanner DB. This parameter is mainly used for infrastructure nodes.
 
+| `scanner.db.hostAliases`
+| Use this parameter to inject hosts and ips into the pod's hosts file.
```
Comment (kcarmichael08) [inline modules/central-configuration-options-operator.adoc:166]:
```suggestion
| Use this parameter to inject hosts and IP addresses into the pod's hosts file.
```

**Review by kcarmichael08 — COMMENTED** [review]
Thank you! This looks good - I had a couple of comments. Unfortunately we had to change the main RHACS branch to `rhacs-docs-main`, so if you can change the parent branch that would be great. (I hope that doesn't cause too much work.)

**Review by kcarmichael08 — APPROVED** [review]
lgtm

**Comment by kcarmichael08** [comment]
Do not merge into 4.4 until 4.4.3 patch is released.

**Pull request merged**